streaming/rrdpush.c

```diff
@@ -641,26 +648,41 @@ static int rrdpush_sender_thread_connect_to_master(RRDHOST *host, int default_po
 
     info("STREAM %s [send to %s]: waiting response from remote netdata...", host->hostname, connected_to);
 
+    ssize_t received;
 #ifdef ENABLE_HTTPS
-    if(recv_timeout(&host->ssl,host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout) == -1) {
+    received = recv_timeout(&host->ssl,host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout);
+    if(received == -1) {
 #else
-    if(recv_timeout(host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout) == -1) {
+    received = recv_timeout(host->rrdpush_sender_socket, http, HTTP_HEADER_SIZE, 0, timeout);
+    if(received == -1) {
 #endif
         error("STREAM %s [send to %s]: remote netdata does not respond.", host->hostname, connected_to);
         rrdpush_sender_thread_close_socket(host);
         return 0;
     }
 
-    int answer = strncmp(http, START_STREAMING_PROMPT_V2, strlen(START_STREAMING_PROMPT_V2));
-    if(!answer) {
-        host->labels_flag |= LABEL_FLAG_UPDATE_STREAM;
-        host->labels_flag &= ~LABEL_FLAG_STOP_STREAM;
+    http[received] = '\0';
+    int answer = -1;
+    char *version_start = strchr(http, '=');
+    if(version_start) {
+        version_start++;
+        answer = memcmp(http, START_STREAMING_PROMPT_VN, (size_t)(version_start - http));
+        if(!answer) {
+            rrdpush_set_flags_to_newest_stream(host);
+            host->stream_version = (uint32_t)strtol(version_start, NULL, 10);
+        }
     } else {
-        answer = strncmp(http, START_STREAMING_PROMPT, strlen(START_STREAMING_PROMPT));
+        answer = memcmp(http, START_STREAMING_PROMPT_V2, strlen(START_STREAMING_PROMPT_V2));
         if(!answer) {
-            host->labels_flag |= LABEL_FLAG_STOP_STREAM;
-            host->labels_flag &= ~LABEL_FLAG_UPDATE_STREAM;
-            info("STREAM %s [send to %s]: is using an old Netdata.", host->hostname,  connected_to);
+            rrdpush_set_flags_to_newest_stream(host);
```

Comment:
There are identical messages that the slave uses an old Netdata but the check has been changed to START_STREAMING_V2

Reply:
The beginning is 100% equal, and this was done to create "message compatibility" (It is between quotes, because I think I generalized the concept ).
The system won't fail when do these kind of tests, because we are using `memcmp` with the length we want to compare.
Finally, we have this order, because I hope, that in a short period of time, nobody will use the oldest Netdata stream protocol.

Reply:
But we are trying to negotiate the highest version that both ends understand?

Reply:
For while yes, but when we begin to move from protocol version 2 in front, we will use the algorithm specified in the issue 7804.